core/src/main/scala/com/pennsieve/core/utilities/Container.scala

```diff
@@ -261,12 +254,34 @@ trait SecureCoreContainer
     } yield ()
 }
 
+object ContainerTypes {
+  type SnsTopic = String
+}
+
 trait StorageContainer {
   self: Container with DatabaseContainer with OrganizationContainer =>
 
   lazy val storageManager = StorageManager.create(self, organization)
 }
 
+trait ChangelogContainer {
+  self: Container with SecureCoreContainer with SNSContainer =>
+
+  val events_topic: SnsTopic =
+    config.as[String]("pennsieve.changelog.sns_topic")
+
+  lazy val changelogManager =
+    new ChangelogManager(db, organization, user, events_topic, sns)
+
+  lazy val datasetPublicationStatusManager: DatasetPublicationStatusManager =
+    new DatasetPublicationStatusManager(
+      db,
+      user,
+      datasetPublicationStatusMapper,
+      changelogManager.changelogEventMapper
+    )
```

Comment:
Does this need to be inside of the `ChangelogContainer`?

Reply:
Yeah, probably better to separate.